cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -298,27 +298,27 @@ rmm::device_uvector<size_type> segmented_count_bits(bitmask_type const* bitmask,
 
   // Allocate temporary memory.
   size_t temp_storage_bytes{0};
-  CUDA_TRY(cub::DeviceSegmentedReduce::Sum(nullptr,
-                                           temp_storage_bytes,
-                                           num_set_bits_in_word,
-                                           d_bit_counts.begin(),
-                                           num_ranges,
-                                           first_word_indices,
-                                           last_word_indices,
-                                           stream.value()));
+  CUDF_CUDA_TRY(cub::DeviceSegmentedReduce::Sum(nullptr,
+                                                temp_storage_bytes,
+                                                num_set_bits_in_word,
+                                                d_bit_counts.begin(),
+                                                num_ranges,
+                                                first_word_indices,
+                                                last_word_indices,
+                                                stream.value()));
   rmm::device_buffer d_temp_storage(temp_storage_bytes, stream);
 
   // Perform segmented reduction.
-  CUDA_TRY(cub::DeviceSegmentedReduce::Sum(d_temp_storage.data(),
-                                           temp_storage_bytes,
-                                           num_set_bits_in_word,
-                                           d_bit_counts.begin(),
-                                           num_ranges,
-                                           first_word_indices,
-                                           last_word_indices,
-                                           stream.value()));
+  CUDF_CUDA_TRY(cub::DeviceSegmentedReduce::Sum(d_temp_storage.data(),
+                                                temp_storage_bytes,
+                                                num_set_bits_in_word,
+                                                d_bit_counts.begin(),
+                                                num_ranges,
+                                                first_word_indices,
+                                                last_word_indices,
+                                                stream.value()));
 
-  CHECK_CUDA(stream.value());
+  CUDF_CHECK_CUDA(stream.value());
```

Comment:
This one seems unnecessary. The `CHECK_CUDA` does a `cudaStreamSynchronize` on the stream.

Reply:
I agree with your assessment, good catch.

Reply:
> The `CHECK_CUDA` does a `cudaStreamSynchronize` on the stream.

Only in debug mode.